src/app/components/calendar/calendar.ts

```diff
@@ -3573,8 +3575,7 @@ export class Calendar implements OnInit, OnDestroy, ControlValueAccessor {
                     `;
                 }
             }
-
-            (<HTMLStyleElement>this.responsiveStyleElement).innerHTML = innerHTML;
+            this.responsiveStyleElement.innerHTML = this.domSanitizer.bypassSecurityTrustStyle(innerHTML) as string;
```

Comment:
If the original PR with textContent was not working **because the assignment value contained markup that wasn’t just the CSS code**, how can we be sure that we can trust this assignment value as a style?

Reply:
bypassSecurityTrustStyle will Bypass security and trust the given value to be safe style value (CSS).
abstract bypassSecurityTrustStyle(value: string): SafeStyle
since the type is assigned to responsiveStyleElement we cannot set innerHTML as SafeStyle.
so the suggestion here is to make the responsiveStyleElement as any and then we do not need the as string cast.
I already merged this suggestion into this pull request.

On the other hand, **these styles are created internally by primeng**, so they are trusted and there is no need for sanitization.
Moreover, **all the inputs from user are SANITIZED** then bypassed by SafeHtml pipe.
```
   constructor(@Inject(PLATFORM_ID) private readonly platformId: any, private readonly domSanitizer: DomSanitizer) { }

    public transform(value: string): SafeHtml {
        if (!value || !isPlatformBrowser(this.platformId)) {
            return value;
        }

+         const sanitizedValue = this.domSanitizer.sanitize(SecurityContext.HTML ,value);
+         return this.domSanitizer.bypassSecurityTrustHtml(sanitizedValue);

    }
```